### PR TITLE
Use GitHub Actions for CI

### PR DESCRIPTION
**Pull request by mathiasvr**
GitHub Actions workflow for running tests and building on all platforms.

Nightly release is not included, but I can add it here or in later PR if preferred.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: .github/workflows/ci.yml
Comment (haslinghuis) [inline .github/workflows/ci.yml:25]: 14.18.2 is LTS

File: .editorconfig
Comment (McGiverGim) [inline .editorconfig:9]: I think this changes don't belong to this PR. If true please remove. No problem if you want to open another PR to add this.

**Review by McGiverGim — CHANGES_REQUESTED** [review]
I see changes in files that don't seem to belong to the PR (.editorconfig,)

File: .github/workflows/ci.yml
Comment (McGiverGim) [reply .github/workflows/ci.yml:25]: I suppose 14.18 is the LTS. If some bug is found in the LTS version a 14.18.3 will be released. Another thing is that if we can use the same version that we publish in:

https://github.com/betaflight/betaflight-configurator/blob/e43025cffe50eb1453f2da7977506f9a6064c8ff/.nvmrc#L1

but this implies to remember to modify both files at the same time.

**Comment by McGiverGim** [comment]
> Nightly release is not included, but I can add it here or in later PR if preferred.

I suppose is better to add it here, if not, we can't remove the Azure part. With only this, both builds are fired.

File: .editorconfig
Comment (mathiasvr) [reply .editorconfig:9]: Ah yes this is from #2734, I will remove from this PR.

File: .github/workflows/ci.yml
Comment (mathiasvr) [reply .github/workflows/ci.yml:25]: Yes 14.18 is the same as 14.18.x, so it will get the latest patch. The LTS is simply Node 14 (same as 14.x.x) with minor version updates as well. I didn't know you used nvm, i will look into using that version instead. However, unless we have a good reason for locking even the patch version, I would consider using at least 14.18.x since patches are non-breaking and could be important bug/security fixes.

Comment (haslinghuis) [reply .github/workflows/ci.yml:25]: Thanks for the heads up

Comment (mathiasvr) [reply .github/workflows/ci.yml]: Right, it can make sense to update the version manually for reproducible builds, but this node.js version is not a dependency of the executable but a tool used to "compile" it and patches are non-breaking and should not change the output of the build (but may change how it's build). Anyway I don't really mind fixing the version. I have updated PR to use .nvmrc version.

File: .github/workflows/ci.yml
Comment (McGiverGim) [inline .github/workflows/ci.yml:97]: I suppose this retention days are only to store them in Github Actions? If we "release" a nightly for example in the betaflight-configurator-nightlies repo it does not affect, true?

Comment (mathiasvr) [reply .github/workflows/ci.yml:97]: Good catch! I reduced this for testing, but it might be better to keep the pull request artifacts longer?
The artifacts uploaded to GitHub releases are not removed.

Comment (McGiverGim) [reply .github/workflows/ci.yml:97]: If posible at least one week or a little more. If you find a way to attach or link to the artifacts in a response to the PR, we can't revisit this. But by now, I think 10-14 days is enough.

Comment (mathiasvr) [reply .github/workflows/ci.yml]: Okay I updated it to 10 days.

Comment (haslinghuis) [reply .github/workflows/ci.yml]: 10 days is definite better than 7 days yes. Don't know about storage capacity we have.

**Review by haslinghuis — DISMISSED** [review]

File: .github/workflows/ci.yml
Comment (McGiverGim) [reply .github/workflows/ci.yml]: I think we don't have storage limit. GitHub let's store until 400 days I think (default 90 days).

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by McGiverGim — DISMISSED** [review]

**Review by haslinghuis — APPROVED** [review]

**Review by McGiverGim — APPROVED** [review]

**Review by chmelevskij — APPROVED** [review]

**Pull request merged**
